### PR TITLE
Dialog doesn't close after it is successfully completed #611 : Fixed

### PR DESCRIPTION
**Pull request by chaitanyaparashar**
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Apply the `MifosStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/chargedialog/ChargeDialogFragment.java [hunk 1]
```diff
@@ -130,6 +130,7 @@ public void saveNewCharge() {
         chargesPayload.setDateFormat("dd MMMM yyyy");
         chargesPayload.setChargeId(chargeId);
         initiateChargesCreation(chargesPayload);
+        getDialog().dismiss();
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/chargedialog/ChargeDialogFragment.java]: No, This is not right place to dismiss the dialog, do it https://github.com/chaitanyaparashar/android-client/blob/b8cc8a94d2a6c40ef33eaa2309da3535b501f4df/mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/chargedialog/ChargeDialogFragment.java#L196

**Comment by luckyman20** [comment]
@chaitanyaparashar Please squash the commits and fix the travis error.

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/chargedialog/ChargeDialogFragment.java [hunk 2]
```diff
@@ -130,6 +130,7 @@ public void saveNewCharge() {
         chargesPayload.setDateFormat("dd MMMM yyyy");
         chargesPayload.setChargeId(chargeId);
         initiateChargesCreation(chargesPayload);
+
```
Comment (luckyman20) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/chargedialog/ChargeDialogFragment.java]: Delete this line also

**Comment by iml-v** [comment]
@chaitanyaparashar  are you working on this issue ? or i should work on this .

**Comment by droidchef** [comment]
This branch has failed on CI and also has conflicts that need resolution.

**Pull request closed without merging**